### ISSUE TITLE
[Bug]: User::getById('admin') returns "system" user

### ISSUE THREAD

**Issue by BlackbitDevs** [pimcore/pimcore#15051]
### Pimcore version

10.5

### Steps to reproduce

Assume we have 2 users in the `users` table:
<img width="382" alt="Bildschirmfoto 2023-04-27 um 08 53 05" src="https://user-images.githubusercontent.com/8749138/234782917-cd6f5cac-7a2c-4efd-b2ee-0a54f69f118a.png">

Run the following PHP script via `bin/console pimcore:run-script`:
```php
<?php
$user = \Pimcore\Model\User::getById('admin');
var_dump($user->getUsername());
```

Output is `system`.

### Expected / actual behavior

There should user object be returned as there is no user with `id="admin"`

**Comment by blankse** [comment]
@BlackbitDevs What you are looking for is `User::getByName('admin')`
The `getById()` method only accepts an integer for the id column in the database table. In Pimcore 11, it will throw an exception, I guess. I think here it will be cast to an integer, so it is 0 and returns the system user.

**Comment by BlackbitDevs** [comment]
@blankse Sorry, I described the wrong expected behaviour. In my real script I had this code
```php
$user = User::getById($value);
if (!$user instanceof User) {
    $user = User::getByName($value);
}
```
And when this gets executed with `$value="admin"` (or I think any non-numerical string) `$user` will always be the `system` user.

Ah, the cause is the type-cast in https://github.com/pimcore/pimcore/blob/adfe3f93bf22696cfe7cecf6b3a4e8e39b7556c9/models/User/AbstractUser/Dao.php#L33

In my app I now changed the order to 
```php
$user = User::getByName($value);
if (!$user instanceof User) {
    $user = User::getById($value);
}
```
then it works.

**Comment by blankse** [comment]
@BlackbitDevs I think you should check after a numeric value and cast it to an integer for getById to be safe for Pimcore 11:
```
$user = User::getByName($value);
if (!$user instanceof User && is_numeric($value)) {
    $user = User::getById((int) $value);
}
```

**Comment by BlackbitDevs** [comment]
Ok will do. Would nevertheless leave this issue open because imho current behaviour in Pimcore 10 is wrong.

**Comment by blankse** [comment]
@BlackbitDevs I created a PR: https://github.com/pimcore/pimcore/pull/15086

**Comment by blankse** [comment]
Is fixed by #15086
